### PR TITLE
[activation] LeakyReLU

### PR DESCRIPTION
**Pull request by KSTconsult**

### CODE REVIEW

File: src/runtime/cpu/kernels/kernels.cpp [hunk 1]
```diff
@@ -46,6 +46,17 @@ void deepworks::CPUReLUInputGrad(ConstVector input,
     grad_input = (input.array() > 0.f).select(grad_output, 0.f);
 }
 
+void deepworks::CPULeakyReLUForward(ConstVector X, Vector result, float alpha) {
+    result = (X.array() < 0.f).select(X.array() * alpha, X);
+}
+
+void deepworks::CPULeakyReLUInputGrad(ConstVector input, ConstVector grad_output,
+                                      Vector grad_input, float alpha) {
+    Eigen::MatrixXf ones = Eigen::MatrixXf::Ones(input.outerSize(), input.innerSize());
```
Comment (KSTconsult) [inline src/runtime/cpu/kernels/kernels.cpp:55]: Someone knows how to do it better?

Comment (TolyaTalamanov) [reply src/runtime/cpu/kernels/kernels.cpp:55]: What's the problem ?

**Review by TolyaTalamanov — APPROVED** [review]
@l-bat Have a look

File: src/runtime/cpu/kernels/kernels.cpp (hunk 1 above)
Comment (KSTconsult) [reply src/runtime/cpu/kernels/kernels.cpp:55]: I have to create ones matrix every time

File: src/runtime/cpu/kernels/kernels.cpp [hunk 2]
```diff
@@ -46,6 +46,17 @@ void deepworks::CPUReLUInputGrad(ConstVector input,
...
+    grad_input = grad_output.array() *
+                 (input.array() < 0.f).select(alpha, ones).array();
```
Comment (l-bat) [inline src/runtime/cpu/kernels/kernels.cpp:57]: Can we do this without creating a ones matrix?
```cpp
    grad_input = input.array() < 0.f).select(alpha * grad_output, grad_output).array();
```

Comment (KSTconsult) [reply src/runtime/cpu/kernels/kernels.cpp]: Done.

**Review by l-bat — APPROVED** [review]

**Comment by TolyaTalamanov** [comment]
Thanks! :+1:

**Pull request merged**